keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -675,17 +724,22 @@ class SqlArtifactRepository(
           }, { (version, _, _) ->
             version
           })
+
+        val currentVersion = versions[CURRENT]?.firstOrNull()
         ArtifactVersions(
           name = artifact.name,
           type = artifact.type,
           statuses = releaseStatuses,
           versions = ArtifactVersionStatus(
-            current = versions[CURRENT]?.firstOrNull(),
+            current = currentVersion,
             deploying = versions[DEPLOYING]?.firstOrNull(),
-            pending = versions[PENDING] ?: emptyList(),
+            // take out stateful constraint values that will never happen
+            pending = removeLowerIfCurrentExists(artifact, currentVersion, versions[PENDING]),
```

Comment:
This is pretty confusing. The comment and function name seem disconnected. I'm not sure what to suggest, but wanted to point it out because I think it'll be hard for anyone maintaining this code to understand.

Reply:
Maybe `higherThanCurrent` to follow the pattern you used naming the other function? I think that would make much more sense.

Reply:
I updated all these with gal's suggesting of wording above - I agree it's confusing.